catalog/spatial/geocoding/spatial-geocoding-feature/src/main/java/org/codice/ddf/spatial/geocoding/feature/CatalogFeatureIndexer.java

```diff
@@ -114,33 +122,42 @@ private void createOrUpdateMetacardForFeature(SimpleFeature feature, boolean cre
   }
 
   private Metacard createMetacardForFeature(SimpleFeature feature) throws FeatureIndexingException {
-    Metacard metacard = new MetacardImpl();
+    Metacard metacard = new MetacardImpl(metacardType);
     String countryCode = feature.getID();
-    metacard.setAttribute(new AttributeImpl(Core.TITLE, countryCode));
 
-    String wkt = WKT_WRITER_THREAD_LOCAL.get().write((Geometry) feature.getDefaultGeometry());
-    metacard.setAttribute(new AttributeImpl(Core.LOCATION, wkt));
+    Object nameObject = feature.getAttribute(NAME_KEY);
+    if (nameObject instanceof String) {
+      metacard.setAttribute(new AttributeImpl(Core.TITLE, (String) nameObject));
+
+      metacard.setAttribute(new AttributeImpl(Location.COUNTRY_CODE, countryCode));
+      String wkt = WKT_WRITER_THREAD_LOCAL.get().write((Geometry) feature.getDefaultGeometry());
+      metacard.setAttribute(new AttributeImpl(Core.LOCATION, wkt));
```

Comment:
nit: should we do any validation on the `wkt` before adding it to the metacard?

Reply:
I don't think we do, for malformed geometries, `getDefaultGeometry` would just return null